materialize-boilerplate/validate.go

```diff
@@ -262,54 +241,17 @@ func (v Validator) validateMatchesExistingResource(
 				Type:   pm.Response_Validated_Constraint_FIELD_FORBIDDEN,
 				Reason: "Cannot add a new key location to the field selection of an existing non-delta-updates materialization witout backfilling",
 			}
-		} else if ambiguousFields := v.ambiguousFields(p, boundCollection.Projections); len(ambiguousFields) > 0 {
-			// Projections that would result in ambiguous materialized fields are forbidden if a
-			// different ambiguous projection has already been selected, or optional if none have
-			// yet been selected.
 			if lastBinding != nil && slices.Contains(lastBinding.FieldSelection.AllFields(), p.Field) {
-				// This field has already been selected as the ambiguous field to materialize.
 				if existingField := existingResource.GetField(p.Field); existingField != nil {
-					// If the field already exists in the destination, make sure
-					// the new projection is compatible.
 					if c, err = v.constraintForExistingField(boundCollection, p, *existingField, fieldConfigJsonMap); err != nil {
-						return nil, fmt.Errorf("evaluating constraint for existing ambiguous field: %w", err)
+						return nil, fmt.Errorf("evaluating constraint for existing field: %w", err)
 					}
 				} else {
 					c = &pm.Response_Validated_Constraint{
 						Type:   pm.Response_Validated_Constraint_LOCATION_RECOMMENDED,
```

Comment:
for later: [LOCATION_RECOMMENDED is no longer a thing](https://github.com/estuary/flow/blob/c527020646fa546b442967566a76507a9e79c655/go/protocols/materialize/materialize.proto#L272)

Update to be FIELD_OPTIONAL?